feat: format sats and fiat with current locale

lisabaut opened the pull request:
### Describe the changes you have made in this PR

This PR does:
- passes current locale to `getFormattedFiat` 
- passes current locale to `getFormattedNumber` 
- passes current locale to `getFormattedSats` 
- moves these helpers to SettingsContext
- renders components in tests with SettingsProvider where possible to test the correct number formatting
- add jest global matchMedia config to render tests with SettingsProvider

**Please follow the commit history if you have problems comparing the diff.**

Please checkout this PR yourself and check if I missed a spot. 🙏 

## NOTEs

~I am still thinking of moving `getFormattedNumber` and `getFormattedSats` to the SettingsContext.
(Same as `getFormattedFiat`) **WDYT?**~
=> done ✅ 

### Link this PR to an issue [optional]

Closes https://github.com/getAlby/lightning-browser-extension/issues/1022

### Type of change

- `feat`: New feature (non-breaking change which adds functionality)

### Screenshots of the changes [optional]

<img width="742" alt="fomatted" src="https://user-images.githubusercontent.com/11243967/201914246-24e8327a-fb68-415d-a175-89c68cb76d6b.png">


### How has this been tested?
- added more tests
- checked UI 

### Checklist

- [x] My code follows the style guidelines of this project and performed a self-review of my own code
- [x] New and existing tests pass locally with my changes
- [ ] I checked if I need to make corresponding changes to the documentation (and made those changes if needed)

escapedcat commented:
ack

> I am still thinking of moving getFormattedNumber and getFormattedSats to the SettingsContext.

Curious about the reason. To have all this in the same place. No need to have it in a currency-file?

Just for my understanding:
- Currency is formatted according to the locale-money formatting
- Sats are formatted to the locale-number formatting

Yes? Asking mainly because for a moment I was confused why both amounts are differently formatted. But I guess that makes sense because sats is just a number?

lisabaut commented:
> Curious about the reason. To have all this in the same place. No need to have it in a currency-file?
> 
The actual function would still be in the currency-file.
But the component import it from the context.
Therefore, the components do not need to know about the locale no more.
Same implementation as for getFiat => https://github.com/getAlby/lightning-browser-extension/blob/master/src/app/context/SettingsContext.tsx#L8


> Just for my understanding:
> 
>     * Currency is formatted according to the locale-money formatting
> 
>     * Sats are formatted to the locale-number formatting
> 
> 
> Yes? Asking mainly because for a moment I was confused why both amounts are differently formatted. But I guess that makes sense because sats is just a number?

This is how INTL formats numbers with or without floats according to the language:
- 3190505 => 3,190,505 (Swedish => 3 190 505), this is a number without decimals
- 910.10 => 910.10 (Swedish 910,10), a number with decimals

Maybe the commas for big numbers confused you? 
UK, USA ... they use commas to separate large number (3 digits) and a point for decimals.
In most Europe countries it is the other way round.
Nice read: https://www.languageediting.com/number-formatting-europe-vs-us/

lisabaut commented:
> 

As discussed:

I moved now all formatting helpers to SettingsContext.
And adjusted tests to render with SettingsProvider to test that the formatting still works.

bumi commented on src/app/screens/ConfirmPayment/index.tsx:
```diff
@@ -143,7 +149,7 @@ function ConfirmPayment() {
               <div className="mb-4 p-4 shadow bg-white dark:bg-surface-02dp rounded-lg">
                 <PaymentSummary
-                  amount={invoice.satoshis}
+                  amount={invoice.satoshis || "0"} // how come that sathoshis can be undefined, bolt11?
```
likely those are zero-amount invoices? which bolt11 invoice did you use?

bumi reviewed (commented):
right now we assume the balances are in sats, so we can do `getFormattedFiat(balance)` 
What do we need to change when we work on the task to allow the account/connector decide which currency it is in?

is the settings context then still the right place?

We probably then check in the account the currency of the account and then decide which formatting function we call to display it?

bumi commented:
Re fiat amounts: https://github.com/getAlby/lightning-browser-extension/issues/1742#issuecomment-1310269008 
we tackle that next.
